src/main/scala/uk/gov/nationalarchives/aws/utils/Clients.scala

```diff
@@ -65,7 +65,7 @@ object Clients {
     val httpClient = NettyNioAsyncHttpClient.builder.build
     SfnAsyncClient.builder
       .region(Region.EU_WEST_2)
-      .endpointOverride(URI.create(configFactory.getString("sfn.endpoint")))
+      .endpointOverride(URI.create(configFactory.getString("stepFunction.endpoint")))
```

Comment:
I hadn't noticed this before, but I don't think this library should be reading things directly from an applications config file. A library shouldn't make assumptions about how the client application is storing its config. And it couples the library and the config - if you want to change the structure of the config, you'll have to update the library too.

Can the code pass this value to the `sfnAsyncClient` instead?

Reply:
Yes. I will leave the other clients as is, and raise a JIRA tech debt card, as it'll be quite a bit of work to track down and update the repos that use the other clients.

Reply:
Thanks! I think you're right that it makes sense to create a card to come back to the others.

Reply:
Raised JIRA ticket to fix up the other clients: https://national-archives.atlassian.net/browse/TDR-900